test/util/server/server.go

```diff
@@ -395,6 +401,45 @@ func serviceAccountSecretsExist(client *kclient.Client, namespace string, sa *ka
 	return foundTokenSecret && foundDockercfgSecret
 }
 
+// WaitForPodCreationServiceAccounts ensures that the service account needed for pod creation exists
+// and that the cache for the admission control that checks for pod tokens has caught up to allow
+// pod creation.
+func WaitForPodCreationServiceAccounts(client *kclient.Client, namespace string, serviceAccounts []string) error {
+	if err := WaitForServiceAccounts(client, namespace, serviceAccounts); err != nil {
+		return err
+	}
+
+	testPod := &kapi.Pod{}
+	testPod.GenerateName = "test"
+	testPod.Spec.Containers = []kapi.Container{
+		{
+			Name:  "container",
+			Image: "openshift/origin-pod:latest",
+		},
+	}
+
+	return wait.PollImmediate(time.Second, PodCreationWaitTimeout, func() (bool, error) {
+		pod, err := client.Pods(namespace).Create(testPod)
+		if err != nil {
+			if !kapierrors.IsForbidden(err) {
+				return false, err
+			}
+			// Only continue trying if the reason is that the API token was not found
+			msg := err.(kapierrors.APIStatus).Status().Message
+			if !strings.Contains(msg, "no API token found for service acccount") {
```

Comment:
> I would. I also wonder about removing the service account aspect of this, especially since you're not using the given service accounts in the test pod.

that's probably an oversight more than intent.